### DISCUSSION TITLE
关于自动选人的问题

### DISCUSSION THREAD

**Discussion by ghost** [HSLix/LixAssistantLimbusCompany#98] [General]
在使用体系队时，发现自动选人有时会产生选择错误的情况，但是全程正常运行，没有报错。
可能是因为版本更新无法检测是否满人，从而一直在选。
能否设计成直接不另外选人们，视频在这里：https://www.bilibili.com/video/BV1LDYteEE6s/

**Comment by yblkksk** [comment]
我这也遇到了同样的问题，它会按照sinner_order.txt设置的前六个顺序选择点击六个，恰好把默认选择的六位罪人全取消了选择，愣了一段时间后它会把配置文件里面设置的第七到第十二位给选上开始战斗，这样一来就导致一场战斗的阵容是前六位罪人，另一场战斗是第七到第十二位罪人这样循环。
以及希望可以增加楼层卡包的黑名单功能，有几个关底boss自动p战斗必定会翻车，比如人皮启示录。

**Reply by yblkksk** [reply]
哦还有网络异常时的自动点击重连

**Comment by ghost** [comment]
想到一个主意，能否设置成选人之前点击一次清空选人

**Comment by HSLix** [comment]
感谢反馈，这都是月计总是喜欢折腾选人的 UI 导致的，和 #91 一样，解决方法也一样

至于根本解决方法，还得引入 OCR

关于优化自动选人的流程，你说的清空选人后再选确实是个可行的思路，嘛，为了能尽早让 LALC 刷完罢

**Reply by ghost** [reply]
关于#91的情况，能解决大部分情况，但是新的本里面引入了12/12这个估计还会出现相同的问题

**Reply by HSLix** [reply]
> 关于#91的情况，能解决大部分情况，但是新的本里面引入了12/12这个估计还会出现相同的问题

所以暂时只能治标不治本啊，根本解决才行

**Comment by IWhiteMouseI** [comment]
我在新版本也遇到了同样的问题，有时会只选到两三个人上场，感觉是因为新版本清空选人多了一个确认按钮导致了前几下点击被按钮挡住，增加开始选人的间隔或者多一个点击确认清空的步骤应该能解决？

**Reply by HSLix** [reply]
> 我在新版本也遇到了同样的问题，有时会只选到两三个人上场，感觉是因为新版本清空选人多了一个确认按钮导致了前几下点击被按钮挡住，增加开始选人的间隔或者多一个点击确认清空的步骤应该能解决？

你是穿越过来的人吗？听你的描述仿佛是对旧版本说的，3.0.0后新版本没有这些问题吧，如果出现了，请提供对应录像和日志

**Reply by IWhiteMouseI** [reply]
https://github.com/user-attachments/assets/6502cd21-7e06-4c59-8130-63c3e8e1d608
确实有这个问题...多一个点击的步骤就能解决。
[25-04-12_task_debug.log](https://github.com/user-attachments/files/19711666/25-04-12_task_debug.log)
[25-04-12_task_error.log](https://github.com/user-attachments/files/19711667/25-04-12_task_error.log)
[25-04-12_task_info.log](https://github.com/user-attachments/files/19711668/25-04-12_task_info.log)
[25-04-12_task_warning.log](https://github.com/user-attachments/files/19711669/25-04-12_task_warning.log)

**Reply by HSLix** [reply]
> https://github.com/user-attachments/assets/6502cd21-7e06-4c59-8130-63c3e8e1d608 确实有这个问题...多一个点击的步骤就能解决。 [25-04-12_task_debug.log](https://github.com/user-attachments/files/19711666/25-04-12_task_debug.log) [25-04-12_task_error.log](https://github.com/user-attachments/files/19711667/25-04-12_task_error.log) [25-04-12_task_info.log](https://github.com/user-attachments/files/19711668/25-04-12_task_info.log) [25-04-12_task_warning.log](https://github.com/user-attachments/files/19711669/25-04-12_task_warning.log)

其实有的，只不过LALC是采用 Enter键确认的方式，或许是你的设备远远好于陆爻齐，过于流畅导致的问题，后续会调整相关逻辑，尽可能确保十分流畅的情况也能正常也行：）

**Reply by HSLix** [reply]
但是很可惜录屏不是使用陆爻齐的录像，没法精确定位问题发生的时间点，得等下次更新在试试罢

**Reply by HSLix** [reply]
> https://github.com/user-attachments/assets/6502cd21-7e06-4c59-8130-63c3e8e1d608 确实有这个问题...多一个点击的步骤就能解决。 [25-04-12_task_debug.log](https://github.com/user-attachments/files/19711666/25-04-12_task_debug.log) [25-04-12_task_error.log](https://github.com/user-attachments/files/19711667/25-04-12_task_error.log) [25-04-12_task_info.log](https://github.com/user-attachments/files/19711668/25-04-12_task_info.log) [25-04-12_task_warning.log](https://github.com/user-attachments/files/19711669/25-04-12_task_warning.log)

现在的版本还会出现问题吗：？

**Reply by IWhiteMouseI** [reply]
现在没问题了，感谢大佬。
